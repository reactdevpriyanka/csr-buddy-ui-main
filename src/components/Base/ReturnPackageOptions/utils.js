export const LABEL_CHOICES = [
  {
    label: '1 Label',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Labels',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Labels',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Labels',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Labels',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Labels',
    enabled: true,
    visible: true,
    value: 6,
  },
];

export const BOX_CHOICES = [
  {
    label: '1 Box',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Boxes',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Boxes',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Boxes',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Boxes',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Boxes',
    enabled: true,
    visible: true,
    value: 6,
  },
];
