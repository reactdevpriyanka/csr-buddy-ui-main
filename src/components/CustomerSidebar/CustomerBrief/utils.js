export const isWeight = (name) => name === 'WEIGHT_LIMIT';

export const blacklistedTags = new Set(['DISCOUNT_ABUSER', 'RETURNS_ABUSER', 'SUSPECTED_FRAUD']);
