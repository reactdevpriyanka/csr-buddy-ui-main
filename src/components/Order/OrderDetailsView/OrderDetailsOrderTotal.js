import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Typography, FormLabel } from '@material-ui/core';
import { Card, Divider } from '@mui/material';
import cn from 'classnames';
import { currencyFormatter } from '@/utils/string';
import useOrderDetailsContext from '@/hooks/useOrderDetailsContext';
import TooltipPrimary from '@/components/TooltipPrimary';
import { useMemo } from 'react';
import useAthena from '@/hooks/useAthena';
import OrderAdjustmentDetails from './OrderAdjustmentDetails';

const useStyles = makeStyles((theme) => ({
  root: {
    '& li:first-of-type': {
      whiteSpace: 'nowrap',
    },
  },
  orderDetailContent: {
    padding: theme.utils.fromPx(16),
  },
  title: {
    fontFamily: 'Roboto',
    fontWeight: '700',
    fontSize: theme.utils.fromPx(16),
    paddingBottom: theme.utils.fromPx(12),
    color: theme.palette.blue.dark,
  },
  orderDetailHeader: {
    display: 'inline-flex',
  },
  detail: {
    '&:not(:last-child)': {
      marginBottom: `${theme.utils.fromPx(10)}`,
    },
  },
  value: {
    height: theme.utils.fromPx(20),
    paddingBottom: theme.spacing(0.5),
    marginBottom: theme.spacing(0.5),
    color: '#333333',
    fontWeight: '400',
  },
  right: {
    float: 'right',
  },
  label: {
    fontSize: theme.utils.fromPx(14),
    lineHeight: theme.utils.fromPx(21),
    color: '#666666',
    fontWeight: 700,
  },
  grandTotal: {
    paddingTop: theme.utils.fromPx(8),
  },
  coloradoRetailDeliveryFee: {
    textDecoration: 'underline',
  },
  crdfTooltip: {
    display: 'inline',
  },
}));

const OrderDetailsOrderTotal = ({ expand = false }) => {
  const classes = useStyles();

  const { getLang } = useAthena(); // athena config

  const {
    total = {},
    totalProduct = {},
    totalAdjustment = {},
    totalBeforeTax = {},
    totalSalesTax = {},
    totalShipping = {},
    promotions = [],
    orderFees = [],
  } = useOrderDetailsContext();

  const coloradoRetailDeliveryFee = useMemo(() => {
    const fee = orderFees.find(({ type }) => type === 'RDF_CO');
    return fee?.amount?.value;
  }, [orderFees]);

  return (
    <div data-testid="orderDetailsOrderTotal" className={classes.root}>
      <Card elevation={0}>
        <div className={classes.orderDetailContent}>
          <Typography className={classes.title} variant="h2">
            {getLang('orderTotals', { fallback: 'Order Totals' })}
          </Typography>
          <div className={cn(classes.detail)}>
            <span>
              <FormLabel
                component="label"
                data-testid="order-itemSubtotal-label"
                className={cn(classes.label)}
              >
                {getLang('orderItemsSubtotal', { fallback: 'Item(s) Subtotal:' })}
              </FormLabel>
            </span>
            <span className={classes.right}>
              <FormLabel
                component="label"
                data-testid="order-itemSubtotal-value"
                className={cn(classes.value)}
              >
                {currencyFormatter(totalProduct?.value)}
              </FormLabel>
            </span>
          </div>
          <div className={cn(classes.detail)}>
            <FormLabel component="label" data-testid="order:adjustmentApplied-label">
              <OrderAdjustmentDetails
                expand={expand}
                disableAdjustments={promotions?.length === 0}
                promotions={promotions}
                adjustmentTotal={currencyFormatter(totalAdjustment?.value)}
              />
            </FormLabel>
          </div>
          <div className={cn(classes.detail)}>
            <span>
              <FormLabel
                component="label"
                data-testid="order:shipping-label"
                className={cn(classes.label)}
              >
                {getLang('orderShipping', { fallback: 'Shipping:' })}
              </FormLabel>
            </span>
            <span className={classes.right}>
              <FormLabel
                component="label"
                data-testid="order:shipping-value"
                className={cn(classes.value)}
              >
                {currencyFormatter(totalShipping?.value ? totalShipping?.value : '0.00')}
              </FormLabel>
            </span>
          </div>
          <div className={cn(classes.detail)}>
            <span>
              <FormLabel
                component="label"
                data-testid="order:beforeTax-label"
                className={cn(classes.label)}
              >
                {getLang('orderTaxBeforeTax', { fallback: 'Total Before Tax:' })}
              </FormLabel>
            </span>
            <span className={classes.right}>
              <FormLabel
                component="label"
                data-testid="order:beforeTax-value"
                className={cn(classes.value)}
              >
                {currencyFormatter(totalBeforeTax?.value)}
              </FormLabel>
            </span>
          </div>
          <div className={cn(classes.detail)}>
            <span>
              <FormLabel
                component="label"
                data-testid="order:salesTax-label"
                className={cn(classes.label)}
              >
                {getLang('orderSalesTax', { fallback: 'Sales Tax:' })}
              </FormLabel>
            </span>
            <span className={classes.right}>
              <FormLabel
                component="label"
                data-testid="order:salesTax-value"
                className={cn(classes.value)}
              >
                {currencyFormatter(totalSalesTax.value)}
              </FormLabel>
            </span>
          </div>
          {coloradoRetailDeliveryFee && (
            <div className={cn(classes.detail)}>
              <TooltipPrimary
                placement="top"
                title={`Colorado imposes a Retail Delivery Fee of ${currencyFormatter(
                  coloradoRetailDeliveryFee,
                )} per order on the delivery of taxable items`}
                arrow
                className={classes.crdfTooltip}
              >
                <span>
                  <FormLabel
                    component="label"
                    data-testid="order:colorado-retail-delivery-fee-label"
                    className={cn(classes.label, classes.coloradoRetailDeliveryFee)}
                  >
                    {getLang('orderColoradoDeliveryFee', {
                      fallback: 'Colorado Retail Delivery Fee:',
                    })}
                  </FormLabel>
                </span>
              </TooltipPrimary>

              <span className={classes.right}>
                <FormLabel
                  component="label"
                  data-testid="order:salesTax-value"
                  className={cn(classes.value)}
                >
                  {currencyFormatter(coloradoRetailDeliveryFee)}
                </FormLabel>
              </span>
            </div>
          )}
          <Divider sx={{ color: '#94A7C4' }} />
          <div className={cn(classes.detail, classes.grandTotal)}>
            <span>
              <FormLabel
                component="label"
                data-testid="order:grandTotal-label"
                className={cn(classes.label)}
              >
                {getLang('orderGrandTotal', { fallback: 'Grand Total:' })}
              </FormLabel>
            </span>
            <span className={classes.right}>
              <FormLabel
                component="label"
                data-testid="order:grandTotal-value"
                className={cn(classes.value)}
              >
                {currencyFormatter(total.value)}
              </FormLabel>
            </span>
          </div>
        </div>
      </Card>
    </div>
  );
};

OrderDetailsOrderTotal.propTypes = {
  expand: PropTypes.bool,
};

export default OrderDetailsOrderTotal;
