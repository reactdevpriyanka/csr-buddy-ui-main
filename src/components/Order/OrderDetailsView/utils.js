import { getDayDateYearTimeTimezone } from '@/utils';
import { multiply } from 'lodash';

export const AllowableActions = {
  RETURN_ITEMS: 'RETURN_ITEMS',
  SEND_ORDER_INVOICE: 'SEND_ORDER_INVOICE',
  SEND_ORDER_CONFIRMATION: 'SEND_ORDER_CONFIRMATION',
  SEND_ORDER_CANCELED_NOTIFICATION: 'SEND_ORDER_CANCELED_NOTIFICATION',
  ALLOCATE_INVENTORY: 'ALLOCATE_INVENTORY',
  ADD_FULFILLMENT_COMMENT: 'ADD_FULFILLMENT_COMMENT',
  BLOCK_ORDER: 'BLOCK_ORDER',
  FORCE_RELEASE_ORDER: 'FORCE_RELEASE_ORDER',
  PROCESS_ORDER: 'PROCESS_ORDER',
  CANCEL_ORDER: 'CANCEL_ORDER',
  EDIT_SHIPPING_ADDRESS: 'EDIT_SHIPPING_ADDRESS',
  ADD_PROMOTION_CODE: 'ADD_PROMOTION_CODE',
  ADD_PRICE_ADJUSTMENT: 'ADD_PRICE_ADJUSTMENT',
  REMOVE_ITEM: 'REMOVE_ITEM',
  REDUCE_QUANTITY: 'REDUCE_QUANTITY',
  CREATE_PACKAGE: 'CREATE_PACKAGE',
  CONFIRM_SHIPMENT: 'CONFIRM_SHIPMENT',
  RESEND_SHIPMENT_MESSAGE: 'RESEND_SHIPMENT_MESSAGE',
  FORCE_CANCEL_RELEASE: 'FORCE_CANCEL_RELEASE',
  CANCEL_RELEASE: 'CANCEL_RELEASE',
  CANCEL_RETURN: 'CANCEL_RETURN',
  MARK_AS_RECEIVED: 'MARK_AS_RECEIVED',
  CREATE_NEW_LABELS: 'CREATE_NEW_LABELS',
};

export const AllowableActionTypes = {
  ORDER_ACTIONS: 'orderActions',
  RELEASE_ACTIONS: 'releaseActions',
  RETURN_ACTIONS: 'returnActions',
};

export const PackageReleaseTypes = Object.freeze({
  PACKAGE: Symbol('package'),
  RELEASE: Symbol('release'),
});

// format supplemental data attributes to for tooltip text
export const getAttributes = (tag, attributesMap, vetContactInfo, product, rowData, testId) => {
  switch (tag) {
    case 'DISCONTINUED':
      return (
        <span>
          {product?.discontinueDate != null && (
            <span data-testid="tag:discontinueDate" key="discontinueDate">
              <b>Discontinued Date:</b> {getDayDateYearTimeTimezone(product.discontinueDate)}
            </span>
          )}
        </span>
      );
    // case 'OUT_OF_STOCK': Disabling this until Force BackOrder and OOS back in stock dates are available in the api
    //   return (
    //     <span>
    //       {product?.discontinueDate != null && (
    //         <span key="discontinueDate">
    //           <b>Back in Stock Date:</b>
    //         </span>
    //       )}
    //     </span>
    //   );
    // case 'FORCED_BACK_ORDER':
    //   return (
    //     <span>
    //       {product?.discontinueDate != null && (
    //         <span key="discontinueDate">
    //           <b>Available Date:</b>
    //         </span>
    //       )}
    //     </span>
    //   );
    case 'CONNECT_WITH_A_VET':
      return attributesMap ? (
        <span>
          {vetContactInfo?.petName != null && (
            <span data-testid="tag:petName" key="petName">
              Pet Name: {vetContactInfo.petName}
              <br />
            </span>
          )}
          {attributesMap?.reason != null && (
            <span data-testid="tag:appointmentReason" key="appointmentReason">
              Appointment Reason: {attributesMap.reason}
              <br />
            </span>
          )}
          {attributesMap?.id != null && (
            <span data-testid="tag:appointmentId" key="appointmentId">
              Appointment ID: {attributesMap.id}
              <br />
            </span>
          )}
          {attributesMap?.endTime != null && (
            <span data-testid="tag:appointmentEndTime" key="appointmentEndTime">
              Appointment End Time: {`${getDayDateYearTimeTimezone(attributesMap.endTime)}`}
              <br />
            </span>
          )}
        </span>
      ) : null;
    case 'PHARMACY':
      return vetContactInfo ? (
        <span>
          {vetContactInfo?.clinicName != null && (
            <span data-testid="tag:clinicName" key="clinicName">
              Clinic Name: {vetContactInfo.clinicName}
              <br />
            </span>
          )}
          {vetContactInfo?.contactVet != null && (
            <span data-testid="tag:vetContacted" key="vetContacted">
              Vet Contacted: {vetContactInfo.contactVet ? 'Yes' : 'No'}
              <br />
            </span>
          )}
          {vetContactInfo?.vetProfileId != null && (
            <span data-testid="tag:vetProfileId" key="vetProfileId">
              Vet Profile ID: {vetContactInfo.vetProfileId}
              <br />
            </span>
          )}
          {vetContactInfo?.petName != null && (
            <span data-testid="tag:petName" key="petName">
              Pet Name: {vetContactInfo.petName}
              <br />
            </span>
          )}
        </span>
      ) : null;
    case 'DROPSHIP':
      return (
        <span>
          {rowData?.minDeliveryDate != null && rowData?.maxDeliveryDate != null && (
            <span data-testid="tag:minDeliveryDate" key="minDeliveryDate">
              <b>Delivers between:</b> {getDayDateYearTimeTimezone(rowData.minDeliveryDate)} and{' '}
              {getDayDateYearTimeTimezone(rowData.maxDeliveryDate)}
            </span>
          )}
        </span>
      );
    case 'PERSONALIZED':
      const entries = Object.entries(product?.personalizationAttributeMap || {}).sort(
        (obj1, obj2) => {
          return obj1[0].localeCompare(obj2[0]);
        },
      );
      return (
        <span>
          {entries?.map((entry) => {
            return (
              <div key={entry[0]}>
                <span data-testid={`tag:personalized:${entry[0]}`}>
                  <b>{entry[0]}:</b> {entry[1]}
                </span>
              </div>
            );
          })}
        </span>
      );
    default:
      return null;
  }
};

export const generateReturnItemKey = (returnItem) => returnItem.type + '_' + returnItem.state;

// Sums up the total quantity returned for each type of unique return
export const createTotalReturnedQuantities = (returnItems) => {
  const totalReturnedQuantities = {};

  for (const returnItem of returnItems) {
    const key = returnItem.lineItemId;
    const innerKey = generateReturnItemKey(returnItem);

    if (!totalReturnedQuantities[key]) {
      totalReturnedQuantities[key] = {};
    }

    if (!totalReturnedQuantities[key][innerKey]) {
      totalReturnedQuantities[key][innerKey] = 0;
    }

    // When either type of concession fails, the return service gives us back a totalCredit of 0 and no quantity field.
    // Because of this, we treat each concession like a totalCredit of 1 so it can be distributed to shipments like other concessions
    let specialCaseIncrement = 0;
    if (innerKey === 'PRODUCT_CONCESSION_FAILED' || innerKey === 'SHIPPING_CONCESSION_FAILED') {
      specialCaseIncrement = 1;
    }

    const increment =
      returnItem.type === 'PRODUCT_CONCESSION' || returnItem.type === 'SHIPPING_CONCESSION'
        ? returnItem?.totalCredit + specialCaseIncrement
        : returnItem?.quantity + specialCaseIncrement;

    totalReturnedQuantities[key][innerKey] += increment;
  }

  return totalReturnedQuantities;
};

export const orderHasItemsSpreadAcrossShipments = (shipments, currShipment, itemMap = {}) => {
  if (shipments?.length === 1) return false;

  const lineItemIds = Object.keys(itemMap);
  for (const lineItemId of lineItemIds) {
    const otherShipmentLineItemIds = shipments
      ?.filter((shipment) => shipment.id !== currShipment.id)
      ?.map((shipment) => shipment?.shipmentItems)
      ?.flat()
      ?.map(({ lineItemId }) => lineItemId);
    if (otherShipmentLineItemIds?.includes(lineItemId)) return true;
  }
  return false;
};

// When the same lineItemId is found across multiple shipments, we have no way of knowing which returnItems (if any)
// are attributed to any specific item in any specific shipment.
//
// The number of returns processed for a lineItemId will always be less than the total quantity of that lineItemId across all shipments
// so we iterate through the shipments untill we run out of return tags for that lineItemId.
export const assignReturnsToShipments = (shipments, returnItems, itemMap = {}) => {
  const totalReturnedQuantities = createTotalReturnedQuantities(returnItems);

  for (const [index, shipment] of shipments.entries()) {
    for (const shipmentItem of shipment.shipmentItems) {
      const lineItemId = shipmentItem.lineItemId;

      if (totalReturnedQuantities[lineItemId]) {
        const returnTagsToRender = Object.entries(totalReturnedQuantities[lineItemId])
          ?.filter(([returnItemKey, quantityReturned]) => quantityReturned >= 0.001)
          ?.map(([returnItemKey, quantityReturned]) => returnItemKey);

        shipments[index] = {
          ...shipment,
          returnTagsToRender: returnTagsToRender,
        };

        for (const returnItemKey in totalReturnedQuantities[lineItemId]) {
          const returnedQuantity = totalReturnedQuantities[lineItemId][returnItemKey];
          if (returnedQuantity < 0) continue; // No need to continue to compute if there is no returnable quantity left for a lineItemId

          const returnItem = returnItems.find(
            (item) => generateReturnItemKey(item) === returnItemKey,
          );

          if (!returnItem) continue;

          let quantityToDecrement;

          switch (returnItem?.type) {
            case 'PRODUCT_CONCESSION':
              quantityToDecrement = multiply(
                shipmentItem?.quantity * Number.parseFloat(itemMap[lineItemId]?.unitPrice?.value),
              );
              break;
            case 'SHIPPING_CONCESSION':
              quantityToDecrement = returnItem?.totalCredit;
              break;
            case 'REPLACEMENT':
            case 'REFUND':
              quantityToDecrement = shipmentItem?.quantity;
              break;
            default:
              quantityToDecrement = 0;
              break;
          }

          // Subtract the decrement quantity from the quantity of items (or total value of products in concession scenarios)
          // returned to only add returns to be rendered to a shipment if there were enough returns to justify it
          // when the same lineItemId is spread across multiple packages
          totalReturnedQuantities[lineItemId][returnItemKey] =
            returnedQuantity - quantityToDecrement;
        }
      }
    }
  }
};
