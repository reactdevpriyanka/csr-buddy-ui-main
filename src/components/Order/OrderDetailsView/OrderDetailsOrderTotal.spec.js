import { renderWrap } from '@/utils';
import orderDetailsViewData from '__mock__/orderdetails/orderdetailsview';
import { useRouter } from 'next/router';
import OrderDetailsOrderTotal from './OrderDetailsOrderTotal';

const defaultProps = {
  ...orderDetailsViewData,
};

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

useRouter.mockReturnValue({
  query: {
    id: '1075268420',
  },
});

describe('<OrderDetailsOrderTotal />', () => {
  const render = renderWrap(OrderDetailsOrderTotal, { defaultProps });

  test('it should render children', () => {
    expect(render()).toMatchSnapshot();
  });
});
