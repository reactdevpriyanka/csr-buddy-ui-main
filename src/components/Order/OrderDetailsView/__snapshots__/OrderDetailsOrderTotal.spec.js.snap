// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetailsOrderTotal /> it should render children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
        data-testid="orderDetailsOrderTotal"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-xg2nla-MuiPaper-root-MuiCard-root"
        >
          <div
            class="makeStyles-orderDetailContent-2"
          >
            <h2
              class="MuiTypography-root makeStyles-title-3 MuiTypography-h2"
            >
              Order Totals
            </h2>
            <div
              class="makeStyles-detail-5"
            >
              <span>
                <label
                  class="MuiFormLabel-root makeStyles-label-8"
                  data-testid="order-itemSubtotal-label"
                >
                  Item(s) Subtotal:
                </label>
              </span>
              <span
                class="makeStyles-right-7"
              >
                <label
                  class="MuiFormLabel-root makeStyles-value-6"
                  data-testid="order-itemSubtotal-value"
                >
                  $NaN
                </label>
              </span>
            </div>
            <div
              class="makeStyles-detail-5"
            >
              <label
                class="MuiFormLabel-root"
                data-testid="order:adjustmentApplied-label"
              >
                <div
                  class="makeStyles-accordionComponent-12"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-disabled Component-root-24 Component-root-25 css-19fv6u5-MuiPaper-root-MuiAccordion-root"
                    data-testid="autoship:payment:viewdetails:adjustment:accordion"
                  >
                    <div
                      aria-controls="panel1d-content"
                      aria-disabled="true"
                      aria-expanded="false"
                      class="MuiButtonBase-root Mui-disabled MuiAccordionSummary-root Mui-disabled Component-root-26 Component-root-27 css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                      data-testid="autoship:payment:viewdetails:adjustment:accordion:summary"
                      id="panel1d-header"
                      role="button"
                      tabindex="-1"
                    >
                      <div
                        class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                      >
                        <div
                          class="makeStyles-nameValue-19"
                          id="nameValueContainer"
                        >
                          <span
                            class="makeStyles-displayName-20"
                            data-testid="totalAdjustment_name"
                            id="nameContainer"
                          >
                            Adjustment(s) Applied
                          </span>
                          <span
                            class="makeStyles-displayValue-21"
                            data-testid="totalAdjustment_value"
                            id="valueContainer"
                          >
                            $NaN
                          </span>
                        </div>
                      </div>
                      <div
                        class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-v84d5j-MuiSvgIcon-root"
                          data-testid="ArrowForwardIosSharpIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                      style="min-height: 0px;"
                    >
                      <div
                        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                      >
                        <div
                          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                        >
                          <div
                            aria-labelledby="panel1d-header"
                            class="MuiAccordion-region"
                            id="panel1d-content"
                            role="region"
                          >
                            <div
                              class="MuiAccordionDetails-root ForwardRef(AccordionDetails)-root-28 ForwardRef(AccordionDetails)-root-29 css-15v22id-MuiAccordionDetails-root"
                            >
                              <div
                                data-testid="autoship:payment:viewdetails:adjustment:accordion:details"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
            </div>
            <div
              class="makeStyles-detail-5"
            >
              <span>
                <label
                  class="MuiFormLabel-root makeStyles-label-8"
                  data-testid="order:shipping-label"
                >
                  Shipping:
                </label>
              </span>
              <span
                class="makeStyles-right-7"
              >
                <label
                  class="MuiFormLabel-root makeStyles-value-6"
                  data-testid="order:shipping-value"
                >
                  $0.00
                </label>
              </span>
            </div>
            <div
              class="makeStyles-detail-5"
            >
              <span>
                <label
                  class="MuiFormLabel-root makeStyles-label-8"
                  data-testid="order:beforeTax-label"
                >
                  Total Before Tax:
                </label>
              </span>
              <span
                class="makeStyles-right-7"
              >
                <label
                  class="MuiFormLabel-root makeStyles-value-6"
                  data-testid="order:beforeTax-value"
                >
                  $NaN
                </label>
              </span>
            </div>
            <div
              class="makeStyles-detail-5"
            >
              <span>
                <label
                  class="MuiFormLabel-root makeStyles-label-8"
                  data-testid="order:salesTax-label"
                >
                  Sales Tax:
                </label>
              </span>
              <span
                class="makeStyles-right-7"
              >
                <label
                  class="MuiFormLabel-root makeStyles-value-6"
                  data-testid="order:salesTax-value"
                >
                  $NaN
                </label>
              </span>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-10xvm1q-MuiDivider-root"
            />
            <div
              class="makeStyles-detail-5 makeStyles-grandTotal-9"
            >
              <span>
                <label
                  class="MuiFormLabel-root makeStyles-label-8"
                  data-testid="order:grandTotal-label"
                >
                  Grand Total:
                </label>
              </span>
              <span
                class="makeStyles-right-7"
              >
                <label
                  class="MuiFormLabel-root makeStyles-value-6"
                  data-testid="order:grandTotal-value"
                >
                  $NaN
                </label>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
      data-testid="orderDetailsOrderTotal"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-xg2nla-MuiPaper-root-MuiCard-root"
      >
        <div
          class="makeStyles-orderDetailContent-2"
        >
          <h2
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h2"
          >
            Order Totals
          </h2>
          <div
            class="makeStyles-detail-5"
          >
            <span>
              <label
                class="MuiFormLabel-root makeStyles-label-8"
                data-testid="order-itemSubtotal-label"
              >
                Item(s) Subtotal:
              </label>
            </span>
            <span
              class="makeStyles-right-7"
            >
              <label
                class="MuiFormLabel-root makeStyles-value-6"
                data-testid="order-itemSubtotal-value"
              >
                $NaN
              </label>
            </span>
          </div>
          <div
            class="makeStyles-detail-5"
          >
            <label
              class="MuiFormLabel-root"
              data-testid="order:adjustmentApplied-label"
            >
              <div
                class="makeStyles-accordionComponent-12"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-disabled Component-root-24 Component-root-25 css-19fv6u5-MuiPaper-root-MuiAccordion-root"
                  data-testid="autoship:payment:viewdetails:adjustment:accordion"
                >
                  <div
                    aria-controls="panel1d-content"
                    aria-disabled="true"
                    aria-expanded="false"
                    class="MuiButtonBase-root Mui-disabled MuiAccordionSummary-root Mui-disabled Component-root-26 Component-root-27 css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
                    data-testid="autoship:payment:viewdetails:adjustment:accordion:summary"
                    id="panel1d-header"
                    role="button"
                    tabindex="-1"
                  >
                    <div
                      class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
                    >
                      <div
                        class="makeStyles-nameValue-19"
                        id="nameValueContainer"
                      >
                        <span
                          class="makeStyles-displayName-20"
                          data-testid="totalAdjustment_name"
                          id="nameContainer"
                        >
                          Adjustment(s) Applied
                        </span>
                        <span
                          class="makeStyles-displayValue-21"
                          data-testid="totalAdjustment_value"
                          id="valueContainer"
                        >
                          $NaN
                        </span>
                      </div>
                    </div>
                    <div
                      class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-v84d5j-MuiSvgIcon-root"
                        data-testid="ArrowForwardIosSharpIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                      >
                        <div
                          aria-labelledby="panel1d-header"
                          class="MuiAccordion-region"
                          id="panel1d-content"
                          role="region"
                        >
                          <div
                            class="MuiAccordionDetails-root ForwardRef(AccordionDetails)-root-28 ForwardRef(AccordionDetails)-root-29 css-15v22id-MuiAccordionDetails-root"
                          >
                            <div
                              data-testid="autoship:payment:viewdetails:adjustment:accordion:details"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </label>
          </div>
          <div
            class="makeStyles-detail-5"
          >
            <span>
              <label
                class="MuiFormLabel-root makeStyles-label-8"
                data-testid="order:shipping-label"
              >
                Shipping:
              </label>
            </span>
            <span
              class="makeStyles-right-7"
            >
              <label
                class="MuiFormLabel-root makeStyles-value-6"
                data-testid="order:shipping-value"
              >
                $0.00
              </label>
            </span>
          </div>
          <div
            class="makeStyles-detail-5"
          >
            <span>
              <label
                class="MuiFormLabel-root makeStyles-label-8"
                data-testid="order:beforeTax-label"
              >
                Total Before Tax:
              </label>
            </span>
            <span
              class="makeStyles-right-7"
            >
              <label
                class="MuiFormLabel-root makeStyles-value-6"
                data-testid="order:beforeTax-value"
              >
                $NaN
              </label>
            </span>
          </div>
          <div
            class="makeStyles-detail-5"
          >
            <span>
              <label
                class="MuiFormLabel-root makeStyles-label-8"
                data-testid="order:salesTax-label"
              >
                Sales Tax:
              </label>
            </span>
            <span
              class="makeStyles-right-7"
            >
              <label
                class="MuiFormLabel-root makeStyles-value-6"
                data-testid="order:salesTax-value"
              >
                $NaN
              </label>
            </span>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-10xvm1q-MuiDivider-root"
          />
          <div
            class="makeStyles-detail-5 makeStyles-grandTotal-9"
          >
            <span>
              <label
                class="MuiFormLabel-root makeStyles-label-8"
                data-testid="order:grandTotal-label"
              >
                Grand Total:
              </label>
            </span>
            <span
              class="makeStyles-right-7"
            >
              <label
                class="MuiFormLabel-root makeStyles-value-6"
                data-testid="order:grandTotal-value"
              >
                $NaN
              </label>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
